lib/rewards/server/subscribers/world-drop-subscriber.js

```diff
@@ -0,0 +1,164 @@
+    {
+        return closerWalkableNodes.splice(sc.randomInteger(0, closerWalkableNodes.length - 1), 1)[0];
+    }
+
+    static async createRewardRoomObject(roomScene, targetObjectBody, reward)
+    {
+        let dropObjectData = Reward.createDropObjectData(reward, roomScene.roomId);
+        await roomScene.objectsManager.generateObjectFromObjectData(dropObjectData);
+        return await roomScene.roomWorld.createRoomObjectBody(
+            { name: reward.randomRewardId },
+            reward.tileIndex,
+            targetObjectBody.worldTileWidth,
+            targetObjectBody.worldTileHeight,
+            reward.rewardPosition.x,
+            reward.rewardPosition.y
+        );
+    }
+
+    static sendRewardsDataToClient(rewards, client)
+    {
+        let data = this.prepareDataToSend(rewards);
+        // @TODO - BETA - Add new event here for the data.
+        client.broadcast('*', data);
+    }
+
+    static prepareDataToSend(rewards)
+    {
+        let messageData = {
+            [RewardsConst.REWARDS_CONST]: {}
+        };
+        for (let reward of rewards) {
+            messageData[RewardsConst.REWARDS_CONST][reward.randomRewardId + reward.tileIndex] =
+                this.getRewardMessageData(reward);
+        }
+        return messageData;
+    }
+
+    static getRewardMessageData(reward)
+    {
+        return {
+            [RewardsConst.REWARDS_TYPE]: reward.animationData.assetType,
+            [RewardsConst.REWARDS_ASSET_KEY]: reward.animationData.assetKey,
+            [RewardsConst.REWARDS_FILE]: reward.animationData.file,
+            [RewardsConst.REWARDS_PARAMS]: reward.animationData.extraParams,
+            ['x']: reward.rewardPosition.x,
+            ['y']: reward.rewardPosition.y
+        };
+    }
+
+    static getAndValidateParams(params)
+    {
+        let rewardEventData = sc.get(params, 'rewardEventData', false);
+        if (!rewardEventData) {
+            throw new Error('RewardEventData not found on WorldDropSubscriber');
+        }
+        let roomScene = sc.get(params, 'roomScene', false);
+        if (!roomScene) {
+            throw new Error('RoomScene not found on WorldDropSubscriber');
+        }
+        let itemRewards = sc.get(rewardEventData, 'itemRewards', false);
+
+        let targetObjectBody = sc.get(rewardEventData.target_object, 'objectBody', false);
+        if (!targetObjectBody) {
+            throw new Error('target_object objectBody not found on WorldDropSubscriber');
+        }
+        return {
+            roomScene,
+            itemRewards,
+            targetObjectBody
+        };
+    }
+
+    static setRewardTimeDisappear(dropObject, roomScene)
+    {
+        setTimeout(() => {
+            let rewardMessageActions = new RewardMessageActions();
+            rewardMessageActions.removeDrop(dropObject, roomScene);
+        }, roomScene.config.get('server/rewards/actions/disappearTime'));
+    }
+}
+
+module.exports.WorldDropSubscriber = WorldDropSubscriber;
```

Comment:
after looking at the entire class indeed it has too many responsabilities, try to split it a bit into sub-classes and include any required methods into the Pathfinder class itself

Reply:
removed a few methods, tell me how do you see the class after those changes.